### PR TITLE
Add YR redirects

### PR DESCRIPTION
**Pull request by kelsonic**
## Description
**Issue:** https://github.com/department-of-veterans-affairs/va.gov-team/issues/4975

This PR implements the following redirects:

Current URL  |  Redirect to new URL
---  |  ---
www.benefits.va.gov/gibill/yellow_ribbon/yrp_list_2019.asp <br> **Update this existing redirect** | https://www.va.gov/education/yellow-ribbon-participating-schools/
*https://www.benefits.va.gov/gibill/yellow_ribbon/2019/states/ak.asp* | *https://www.va.gov/education/yellow-ribbon-participating-schools/*
*https://www.benefits.va.gov/gibill/yellow_ribbon/2019/states/al.asp* | *https://www.va.gov/education/yellow-ribbon-participating-schools/*
*https://www.benefits.va.gov/gibill/yellow_ribbon/2019/states/ar.asp* | *https://www.va.gov/education/yellow-ribbon-participating-schools/*
*https://www.benefits.va.gov/gibill/yellow_ribbon/2019/states/az.asp* | *https://www.va.gov/education/yellow-ribbon-participating-schools/*
*https://www.benefits.va.gov/gibill/yellow_ribbon/2019/states/ca.asp* | *https://www.va.gov/education/yellow-ribbon-participating-schools/*
*https://www.benefits.va.gov/gibill/yellow_ribbon/2019/states/co.asp* | *https://www.va.gov/education/yellow-ribbon-participating-schools/*
*https://www.benefits.va.gov/gibill/yellow_ribbon/2019/states/ct.asp* | *https://www.va.gov/education/yellow-ribbon-participating-schools/*
*https://www.benefits.va.gov/gibill/yellow_ribbon/2019/states/dc.asp* | *https://www.va.gov/education/yellow-ribbon-participating-schools/*
*https://www.benefits.va.gov/gibill/yellow_ribbon/2019/states/de.asp* | *https://www.va.gov/education/yellow-ribbon-participating-schools/*
*https://www.benefits.va.gov/gibill/yellow_ribbon/2019/states/fl.asp* | *https://www.va.gov/education/yellow-ribbon-participating-schools/*
*https://www.benefits.va.gov/gibill/yellow_ribbon/2019/states/ga.asp* | *https://www.va.gov/education/yellow-ribbon-participating-schools/*
*https://www.benefits.va.gov/gibill/yellow_ribbon/2019/states/hi.asp* | *https://www.va.gov/education/yellow-ribbon-participating-schools/*
*https://www.benefits.va.gov/gibill/yellow_ribbon/2019/states/ia.asp* | *https://www.va.gov/education/yellow-ribbon-participating-schools/*
*https://www.benefits.va.gov/gibill/yellow_ribbon/2019/states/id.asp* | *https://www.va.gov/education/yellow-ribbon-participating-schools/*
*https://www.benefits.va.gov/gibill/yellow_ribbon/2019/states/il.asp* | *https://www.va.gov/education/yellow-ribbon-participating-schools/*
*https://www.benefits.va.gov/gibill/yellow_ribbon/2019/states/in.asp* | *https://www.va.gov/education/yellow-ribbon-participating-schools/*
*https://www.benefits.va.gov/gibill/yellow_ribbon/2019/states/ks.asp* | *https://www.va.gov/education/yellow-ribbon-participating-schools/*
*https://www.benefits.va.gov/gibill/yellow_ribbon/2019/states/ky.asp* | *https://www.va.gov/education/yellow-ribbon-participating-schools/*
*https://www.benefits.va.gov/gibill/yellow_ribbon/2019/states/la.asp* | *https://www.va.gov/education/yellow-ribbon-participating-schools/*
*https://www.benefits.va.gov/gibill/yellow_ribbon/2019/states/ma.asp* | *https://www.va.gov/education/yellow-ribbon-participating-schools/*
*https://www.benefits.va.gov/gibill/yellow_ribbon/2019/states/md.asp* | *https://www.va.gov/education/yellow-ribbon-participating-schools/*
*https://www.benefits.va.gov/gibill/yellow_ribbon/2019/states/me.asp* | *https://www.va.gov/education/yellow-ribbon-participating-schools/*
*https://www.benefits.va.gov/gibill/yellow_ribbon/2019/states/mi.asp* | *https://www.va.gov/education/yellow-ribbon-participating-schools/*
*https://www.benefits.va.gov/gibill/yellow_ribbon/2019/states/mn.asp* | *https://www.va.gov/education/yellow-ribbon-participating-schools/*
*https://www.benefits.va.gov/gibill/yellow_ribbon/2019/states/mo.asp* | *https://www.va.gov/education/yellow-ribbon-participating-schools/*
*https://www.benefits.va.gov/gibill/yellow_ribbon/2019/states/ms.asp* | *https://www.va.gov/education/yellow-ribbon-participating-schools/*
*https://www.benefits.va.gov/gibill/yellow_ribbon/2019/states/mt.asp* | *https://www.va.gov/education/yellow-ribbon-participating-schools/*
*https://www.benefits.va.gov/gibill/yellow_ribbon/2019/states/nc.asp* | *https://www.va.gov/education/yellow-ribbon-participating-schools/*
*https://www.benefits.va.gov/gibill/yellow_ribbon/2019/states/nd.asp* | *https://www.va.gov/education/yellow-ribbon-participating-schools/*
*https://www.benefits.va.gov/gibill/yellow_ribbon/2019/states/ne.asp* | *https://www.va.gov/education/yellow-ribbon-participating-schools/*
*https://www.benefits.va.gov/gibill/yellow_ribbon/2019/states/nh.asp* | *https://www.va.gov/education/yellow-ribbon-participating-schools/*
*https://www.benefits.va.gov/gibill/yellow_ribbon/2019/states/nj.asp* | *https://www.va.gov/education/yellow-ribbon-participating-schools/*
*https://www.benefits.va.gov/gibill/yellow_ribbon/2019/states/nm.asp* | *https://www.va.gov/education/yellow-ribbon-participating-schools/*
*https://www.benefits.va.gov/gibill/yellow_ribbon/2019/states/nv.asp* | *https://www.va.gov/education/yellow-ribbon-participating-schools/*
*https://www.benefits.va.gov/gibill/yellow_ribbon/2019/states/ny.asp* | *https://www.va.gov/education/yellow-ribbon-participating-schools/*
*https://www.benefits.va.gov/gibill/yellow_ribbon/2019/states/oh.asp* | *https://www.va.gov/education/yellow-ribbon-participating-schools/*
*https://www.benefits.va.gov/gibill/yellow_ribbon/2019/states/ok.asp* | *https://www.va.gov/education/yellow-ribbon-participating-schools/*
*https://www.benefits.va.gov/gibill/yellow_ribbon/2019/states/or.asp* | *https://www.va.gov/education/yellow-ribbon-participating-schools/*
*https://www.benefits.va.gov/gibill/yellow_ribbon/2019/states/overseas.asp* | *https://www.va.gov/education/yellow-ribbon-participating-schools/*
*https://www.benefits.va.gov/gibill/yellow_ribbon/2019/states/pa.asp* | *https://www.va.gov/education/yellow-ribbon-participating-schools/*
*https://www.benefits.va.gov/gibill/yellow_ribbon/2019/states/pr.asp* | *https://www.va.gov/education/yellow-ribbon-participating-schools/*
*https://www.benefits.va.gov/gibill/yellow_ribbon/2019/states/ri.asp* | *https://www.va.gov/education/yellow-ribbon-participating-schools/*
*https://www.benefits.va.gov/gibill/yellow_ribbon/2019/states/sc.asp* | *https://www.va.gov/education/yellow-ribbon-participating-schools/*
*https://www.benefits.va.gov/gibill/yellow_ribbon/2019/states/sd.asp* | *https://www.va.gov/education/yellow-ribbon-participating-schools/*
*https://www.benefits.va.gov/gibill/yellow_ribbon/2019/states/tn.asp* | *https://www.va.gov/education/yellow-ribbon-participating-schools/*
*https://www.benefits.va.gov/gibill/yellow_ribbon/2019/states/tx.asp* | *https://www.va.gov/education/yellow-ribbon-participating-schools/*
*https://www.benefits.va.gov/gibill/yellow_ribbon/2019/states/ut.asp* | *https://www.va.gov/education/yellow-ribbon-participating-schools/*
*https://www.benefits.va.gov/gibill/yellow_ribbon/2019/states/va.asp* | *https://www.va.gov/education/yellow-ribbon-participating-schools/*
*https://www.benefits.va.gov/gibill/yellow_ribbon/2019/states/vt.asp* | *https://www.va.gov/education/yellow-ribbon-participating-schools/*
*https://www.benefits.va.gov/gibill/yellow_ribbon/2019/states/wa.asp* | *https://www.va.gov/education/yellow-ribbon-participating-schools/*
*https://www.benefits.va.gov/gibill/yellow_ribbon/2019/states/wi.asp* | *https://www.va.gov/education/yellow-ribbon-participating-schools/*
*https://www.benefits.va.gov/gibill/yellow_ribbon/2019/states/wv.asp* | *https://www.va.gov/education/yellow-ribbon-participating-schools/*
*https://www.benefits.va.gov/gibill/yellow_ribbon/2019/states/wy.asp* | *https://www.va.gov/education/yellow-ribbon-participating-schools/*

## Testing done
N/A

## Screenshots
N/A

## Acceptance criteria
- [x] Implement YR redirects.

## Definition of done
- [x] Events are logged appropriately
- [x] Documentation has been updated, if applicable
- [x] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

**Review by ncksllvn — COMMENTED** [review]
Just wondering - do we, in the end, want to redirect the entire `www.benefits.va.gov/gibill/yellow_ribbon/` directory?

File: src/applications/proxy-rewrite/redirects/crossDomainRedirects.json [hunk 1]
```diff
@@ -723,5 +718,275 @@
     "domain": "www.benefits.va.gov",
     "src": "/gibill/fgib/vettec_veteran.asp",
     "dest": "/education/about-gi-bill-benefits/how-to-use-benefits/vettec-high-tech-program/"
+  },
+  {
+    "domain": "www.benefits.va.gov",
```
Comment (kelsonic) [inline src/applications/proxy-rewrite/redirects/crossDomainRedirects.json:723]: This redirect replaces a previous one: https://github.com/department-of-veterans-affairs/vets-website/pull/12775/files#diff-acef04b7f830ed23ad90ea40684bf634L303

**Comment by ncksllvn** [comment]
Following up on my message, it looks like we want to redirect the whole `https://www.benefits.va.gov/gibill/yellow_ribbon/2019/states/` directory...the way we're doing it here I think is solid, but I was just curious if we'd want to consider adding support for a new redirect pattern for this since there are so many. As in, we could add possibly support for regex, or a new property to indicate that the redirect is greedy similar to the server-side redirects.

**Comment by kelsonic** [comment]
I think we have a ticket to address tech debt in `crossDomainRedirects` and see if we can create a way for us to test the redirects: https://github.com/department-of-veterans-affairs/va.gov-team/issues/7790#issuecomment-612052321

>A testing pattern for cross-domain redirects. I wrote a ticket for that somewhere. We could also probably figure out a way to speed up redirects by separating it out from the rest of the proxy-inject code.

Perhaps we can augment the [current redirect normalization](https://github.com/department-of-veterans-affairs/vets-website/blob/master/src/applications/proxy-rewrite/redirects/index.js#L10) to include support for regex that's similar to the devops implementation! 🙂

**Comment by kelsonic** [comment]
@ncksllvn @stephorkandatx Created a follow-up ticket to refactor cross-domain redirects a bit! 🙂 https://github.com/department-of-veterans-affairs/va.gov-team/issues/9487

**Review by ncksllvn — APPROVED** [review]
LGTM

**Pull request merged**